### ISSUE TITLE
Remove decoration code duplication

### ISSUE THREAD

**Issue by jsumners** [fastify/fastify#1116]
As mentioned in #1114 there is code duplication in the decoration API. There is a primary decoration function: https://github.com/fastify/fastify/blob/d99cd61f246b9d505ae0e0a26819008b65a5afe1/lib/decorate.js#L3-L22

But this is duplicated later at:

https://github.com/fastify/fastify/blob/d99cd61f246b9d505ae0e0a26819008b65a5afe1/lib/decorate.js#L52-L76

Since these request and reply specific decorator functions are out of sight when working on the primary decorate function it is easy to overlook writing the same change to them. There should be one decorate function that handles any prototype passed to it.

**Comment by cemremengu** [comment]
Would it break if the following part `decorate` function is changed to

```
-    Object.defineProperty(this, name, { 
+   Object.defineProperty(this.prototype, name, { 
      get: fn.getter,
      set: fn.setter
    })
```

and then just do 

```
decorate.bind(this)
decorate.bind(this._Reply)
decorate.bind(this._Request)
```

**Comment by jsumners** [comment]
It's possible the change will require touching other parts of the code. I think it is clearer if the function does not rely on `this` being the object manipulated. `decorate(instance, name, value)` (or similar) is preferable in m opinion.

**Comment by cemremengu** [comment]
Landed in #1120

**Comment by cemremengu** [comment]
Ping for closing
